core-bundle/src/Routing/Content/ContentUrlResult.php

```diff
@@ -14,15 +14,14 @@
 
 use Contao\CoreBundle\Exception\ForwardPageNotFoundException;
 use Contao\PageModel;
-use Nyholm\Psr7\Uri;
 
 final class ContentUrlResult
 {
     private bool $redirect = false;
 
     public function __construct(public readonly object|string $content)
     {
-        if (\is_string($content) && !(new Uri($content))->getScheme()) {
+        if (\is_string($content) && !parse_url($content, PHP_URL_SCHEME)) {
```

Comment:
Is this a related change?

Reply:
well I just stumbeled upon this, it's a dependency-free way to do the same.

Reply:
Yes, but `Nyholm\Psr7\Uri` is a complete PSR-7 URI implementation that does more than `parse_url()`. It normalizes URIs to lower case for example. So to stay consistent with the rest of the code base, we should continue to use it.

Reply:
As we already use this in other places like https://github.com/contao/contao/blob/98974328a16e96ebe1b0e789cd256a5f0c8150c0/core-bundle/src/Routing/Content/StringResolver.php#L35 I think it is OK to use this here. To check if there is a schema at all, the two variants are functionally equivalent, but `parse_url()` should be way faster. _(Not sure if speed is relevant for `ContentUrlResult`?)_

Reply:
Do you mean in general or just for the `PHP_URL_SCHEME` case? Because we indeed use `parse_url()` in several other places as well, even to get the hostname:

https://github.com/contao/contao/blob/591a469b595e17a9cf3458cb8b1e54f264228254/news-bundle/src/Migration/FeedMigration.php#L110

I would then check the other uses of `Nyholm\Psr7\Uri`, too.

Reply:
`PHP_URL_HOST` should be fine too, if your use case is also case-insensitive. 
But for port, path, query and fragment we should not use `parse_url()`.

Reply:
I would prefer to consistently use the library, but let‘s discuss this in #6984.